svm/src/transaction_processor.rs

```diff
@@ -240,8 +240,64 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
         self.sysvar_cache.read().unwrap()
     }
 
-    /// Main entrypoint to the SVM.
-    pub fn load_and_execute_sanitized_transactions<CB: TransactionProcessingCallback>(
+    /// Main entrypoint to the SVM. Load and execute a batch of sanitized
+    /// transactions.
+    ///
+    /// The behavior of the SVM is configurable entirely from this method's
+    /// parameters.
+    ///
+    /// * `callbacks`: Reference to an object that implements the
+    ///   `TransactionProcessingCallback` trait. This plugin is used by SVM
+    ///   to load accounts, check account ownership, and more.
+    /// * `sanitized_txs`: A slice of sanitized transactions to process.
+    /// * `environment`: Configures the SVM runtime environment to use when
+    ///   processing the transaction batch.
+    /// * `config`: Additional configurations for controlling the behavior of
+    ///   SVM during transaction processing.
+    ///
+    /// Note: SVM has built-in support for evaluating preprocessing checks that
+    /// may have occurred before invoking the SVM API. These could be useful
+    /// for architectures that may wish to stream batches, performing checks
+    /// on-the-fly that aren't supported by SVM directly.
+    ///
+    /// If you wish to involve pre-processing checks, you can use
+    /// `load_and_execute_batch_with_preprocessing_checks` instead.
+    pub fn load_and_execute_batch<CB: TransactionProcessingCallback>(
+        &self,
+        callbacks: &CB,
+        sanitized_txs: &[impl SVMTransaction],
+        environment: &TransactionProcessingEnvironment,
+        config: &TransactionProcessingConfig,
+    ) -> LoadAndExecuteSanitizedTransactionsOutput {
+        let check_results = vec![
+            Ok(CheckedTransactionDetails {
+                nonce: None,
+                lamports_per_signature: environment.lamports_per_signature
+            });
+            sanitized_txs.len()
+        ];
+        self.load_and_execute_batch_with_preprocessing_checks(
```

Comment:
is there a reason `load_and_execute_batch_with_preprocessing_checks` must take a `Vec`?

Can it take a slice instead, and if it can, then let's avoid a heap allocation here.

Reply:
It can be refactored to take a slice, but that results in cloning of the `NonceInfo` (if any) downstream during `validate_transaction_fee_payer`.

Reply:
ah I see. It takes an owned vec not reference to vec. it consumes the actual vec